python/ray/tests/test_dynres.py

```diff
@@ -11,6 +11,48 @@
 
 logger = logging.getLogger(__name__)
 
+TIMEOUT_RESOURCE_UPDATE = 1  # seconds
+
+
+def retry_until(condition, timeout):
+    # Helper method to retry a condition till it is true or a timeout elapses.
+    # Returns true if the condition was true before the timeout,
+    # False if the timeout elapses.
+    SLEEP_DURATION = 0.1  # seconds
+    time_elapsed = 0
+    while time_elapsed <= timeout:
+        if condition():
+            return True
+        time.sleep(SLEEP_DURATION)
+        time_elapsed += SLEEP_DURATION
```

Comment:
I extracted this method from `test_release_cpus_when_actor_creation_task_blocking`, but I'm not sure if this is the best way to check for a timeout. Should we be using `time_elapsed = start_time - time.time()`? Otherwise this is equivalent to having a fixed number (`timeout/SLEEP_DURATION`) of retries.